synchronization/src/test/java/de/cyface/synchronization/HttpConnectionTest.java

```diff
@@ -118,36 +122,52 @@ public void testGenerateHeader() {
      * <p>
      * This test is indented to avoid MOV-669 where we accidentally used the number of characters in the header
      * instead of the number of bytes to set the fixed content length for steaming mode.
+     *
+     * @throws IOException if the test files were not accessible
      */
     @Test
-    public void testCalculateBytesWrittenToOutputStream() {
+    public void testCalculateBytesWrittenToOutputStream() throws IOException {
 
         // Arrange
         final SyncAdapter.MetaData metaData = new SyncAdapter.MetaData(generateGeoLocation(0), generateGeoLocation(10),
                 "test-did", 78, "test_deviceType", "test_osVersion", "test_appVersion", 10.0, 5, Modality.BICYCLE);
         final String header = oocut.generateHeader(metaData);
 
-        final String fileHeaderPart = generateFileHeaderPart("fileToUpload",
-                "test-did_78." + Constants.TRANSFER_FILE_EXTENSION);
-        final String eventsFileHeaderPart = generateFileHeaderPart("eventsFile",
-                "test-did_78." + Constants.EVENTS_TRANSFER_FILE_EXTENSION);
-
-        final byte[] testFile = "TEST_FÄ`&ô»ω_CONTENT".getBytes(); // with chars which require > 1 Byte
-        final long filePartSize = testFile.length;
+        File testFile = writeTempFile("TEST_FÄ`&ô»ω_CONTENT", "test-did_78", Constants.TRANSFER_FILE_EXTENSION);
```

Comment:
You could add the hint again "chars which require > 1 Byte included" so that it's clear why these characters were selected. This reproduced a Bug we had. I think that there we accidentally used String.length instead of String.toBytes().length to calculate the number of bytes to transmit (or similar). Using those chars makes sure the tests catch this mistake in future.